ludwig/utils/data_utils.py

```diff
@@ -190,12 +198,22 @@ def read_xsv(data_fp, df_lib=PANDAS_DF, separator=",", header=0, nrows=None, ski
     if nrows is not None:
         kwargs["nrows"] = nrows
 
+    # If fetching a preview, create an iterator that will immediately be used to read the first chunk:
+    if "get_preview" in kwargs:
+        kwargs["chunksize"] = kwargs["get_preview"]
+        del kwargs["get_preview"]
+
     try:
         df = df_lib.read_csv(data_fp, **kwargs)
     except ParserError:
         logger.warning("Failed to parse the CSV with pandas default way," " trying \\ as escape character.")
         df = df_lib.read_csv(data_fp, escapechar="\\", **kwargs)
 
+    from collections.abc import Iterator
+
+    if isinstance(df, Iterator):
```

Comment:
Will this condition be satisfied? Seems like the return type of `pd.read_csv` is `TextFileReader` only when `iterator=True`